plugins/neo4j-dump-widget/components/WidgetNeo4jDump.vue

```diff
@@ -0,0 +1,305 @@
+          <div class="col-12 d-flex justify-content-end align-items-end">
+            <b-button type="reset" variant="danger" class="mr-2">Reset</b-button>
+            <span id="disabled-wrapper">
+              <b-button type="submit" :disabled="!neo4jAppIsRunning" variant="primary">Export graph</b-button>
+            </span>
+            <b-tooltip target="disabled-wrapper" v-if="dumpButtonToolTip !== null">{{ dumpButtonToolTip }}</b-tooltip>
+          </div>
+        </div>
+      </div>
+    </b-form>
+  </div>
+</template>
+
+
+<script>
+import bodybuilder from 'bodybuilder'
+import { concat, get, map, random } from 'lodash'
+import { mapState } from 'vuex'
+import { AppStatus } from '../store/Neo4jModule'
+import { default as Neo4jStatusBadge } from '../components/Neo4jStatusBadge.vue'
+import { default as polling } from '../core/mixin/polling'
+
+const SHOULD_START_APP_STATUSES = new Set([AppStatus.Error, AppStatus.Stopped].map(x => AppStatus[x]))
+const CYPHER_SHELL = 'cypher-shell'
+const GRAPHML = 'graphml'
+
+export default {
+  name: 'WidgetNeo4jDump',
+  props: {
+    widget: {
+      type: Object
+    }
+  },
+  mixins: [polling],
+  components: {
+    Neo4jStatusBadge,
+  },
+  data() {
+    return {
+      availableFormats: [
+        { value: null, text: 'Select a format' },
+        { value: CYPHER_SHELL, text: 'Cypher shell' },
+        { value: GRAPHML, text: 'GraphML' },
+      ],
+      dumpFormat: null,
+      fileType: null,
+      fileTypes: [],
+      selectedFileTypes: [],
+      selectedPath: this.$config.get('mountedDataDir') || this.$config.get('dataDir'),
+    }
+  },
+  async created() {
+    this.getFileTypes()
+    await this.resfreshNeo4jAppStatus()
+    await this.startNeo4jAppIfNeed()
+    const fn = this.resfreshNeo4jAppStatus
+    const timeout = () => random(5000, 10000)
+    this.registerPollOnce({ fn, timeout })
+  },
+  unmounted() {
+    this.unregisteredPolls()
+  },
+  computed: {
+    locale() {
+      return this.$i18n.locale
+    },
+    docFields() {
+      return this.getDBSchema()
+    },
+    dumpButtonToolTip() {
+      if (this.neo4jAppStatus === AppStatus.Starting) {
+        return "neo4j extension is starting..."
+      } else if (!this.neo4jAppIsRunning) {
+        return "neo4j extension is not running, refresh this page to start it or wait "
+      }
+      return null
+    },
+    dumpExtension() {
+      if (this.dumpFormat === null) {
+        return null
+      }
+      return this.dumpFormat === GRAPHML ? '.graphml' : '.dump'
+    },
+    project() {
+      return this.$store.state.search.index
+    },
+    neo4jAppIsRunning() {
+      return this.neo4jAppStatus === AppStatus.Running
+    },
+    ...mapState('neo4j', ['neo4jAppStatus'])
+  },
+  methods: {
+    async aggregate(field, name) {
+      let body, options, responses, searchResult
+      let after = null
+      let result = []
+      while (responses === undefined || responses.length === 10) {
```

Comment:
The number 10 seems to be a hard coded value, what does it mean? 
I'm wondering if it would not be safer to use >= instead of === , what do you think ?